src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8467,65 +8467,59 @@ private VariableComponentSyntax ParseDeconstructionComponent(bool topLevel = fal
             }
             else
             {
-                var type = ParseType(false);
+                TypeSyntax type;
+                bool reportMissingType = false;
+                if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken &&
+                    (this.PeekToken(1).Kind == SyntaxKind.CommaToken || this.PeekToken(1).Kind == SyntaxKind.CloseParenToken))
```

Comment:
There are no parse errors in `DeconstructionTests.DeconstructionAssignment`. Consider adding a case to `BadTupleElementTypeInDeconstruction02`.